### PR TITLE
Added AutoOP.

### PR DESCRIPTION
**Pull request by ghost**
Added an auto-op feature in which, if a player joins and isn't opped, they're automatically opped by this feature.

### CODE REVIEW

**Comment by xDestroyer217** [comment]
This should be changed, because we disallow imposters from being opped.

@Scarynoises

**Comment by Scarynoises** [comment]
Oh, right.
I'll quickly cook up a solution,
It will get names from the blockednames.yml file, and if it contains a blocked username it will automatically be blocked.

**Comment by xDestroyer217** [comment]
Alright, but what would the blockednames.yml file would be used for? @Scarynoises

**Comment by Scarynoises** [comment]
The AutoOP class will read the blockednames.yml file and if a player's username contains a blocked name the event will be cancelled.

**Comment by RobinGall2910** [comment]
Seems appealing, alright with me.

**Comment by Scarynoises** [comment]
great.

**Comment by Scarynoises** [comment]
Just pushed an update.
Added support for unauthorized names, file for blacklisted names is blockednames.yml

**Comment by RobinGall2910** [comment]
The question is, I don't know how this would work. There's already a system for when an admin is impostered, so you might need to connect with that. @Scarynoises

**Comment by RobinGall2910** [comment]
Also if you get told or something, please refer to this

https://github.com/TotalFreedom/TotalFreedomMod/wiki/Contributing

I'll make the needed issue

**Comment by Scarynoises** [comment]
I thought you said for imposters that use names such as Deadlox, TrueMU, SkythekidRS or something. My sincere apologies.

**Comment by RobinGall2910** [comment]
Ah, no. Those are obviously fakes, nah, I'm basically talking about the administrators, sorry if I confused you 'that' badly. :P

**Comment by Scarynoises** [comment]
Yeah, you did :p

**Comment by RobinGall2910** [comment]
Then my bad as well. xD

**Comment by PattyToTheWatty** [comment]
Hello, there! Please make an 'issue', and untill it gets accepted, you cant make a pull request.
If it doesn't make it to accepted (e.g. Duplicate, bad idea, etc) you can't make a pull request.

**Comment by RobinGall2910** [comment]
@PattyToTheWatty I just told you...

jesus

**Comment by PattyToTheWatty** [comment]
@RobinGall2910 It's the other way around.

**Comment by RobinGall2910** [comment]
@PattyToTheWatty so? I helped him. You can make it either way.


@Scarynoises just disregard.

File: src/me/StevenLawson/TotalFreedomMod/AutoOP.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Listener;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.player.PlayerJoinEvent;
+import java.io.File;
+
+public class AutoOP implements Listener {
+
```
Comment (rsajeey) [inline src/me/StevenLawson/TotalFreedomMod/AutoOP.java]: Formatting.

File: src/me/StevenLawson/TotalFreedomMod/AutoOP.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    File names = new File(TotalFreedomMod.getPlugin().getDataFolder(), "blockednames.yml");
+    FileConfiguration config = YamlConfiguration.loadConfiguration(names);
+    // We now have no use of the command /opall, yay!
+    @EventHandler
+    public void onJoin(PlayerJoinEvent event) {
+        Player p = event.getPlayer();
+        p.getName();
+        for (String s : config.getStringList("unallowednames")) {
```
Comment (rsajeey) [inline src/me/StevenLawson/TotalFreedomMod/AutoOP.java]: Formatting.

File: src/me/StevenLawson/TotalFreedomMod/TotalFreedomMod.java [hunk 3]
```diff
@@ -74,9 +74,13 @@ public void onLoad()
         setAppProperties();
     }
 
+    public static TotalFreedomMod getPlugin() {
```
Comment (rsajeey) [inline src/me/StevenLawson/TotalFreedomMod/TotalFreedomMod.java]: Formatting.

**Comment by LegendIsAwesomes** [comment]
@scarynoises allmans style please.

**Comment by JeromSar** [comment]
This PR does not meet the [contributing guidelines](https://github.com/TotalFreedom/TotalFreedomMod/wiki/Contributing#pull-request-requirements) and therefore is closed.

**Comment by JeromSar** [comment]
This pull request does not meet the [pull request requirements](https://github.com/TotalFreedom/TotalFreedomMod/wiki/Contributing#pull-request-requirements) and therefore is closed.

**Pull request closed without merging**